File: 1768-Merge-Strings-Alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        word = ""
        p1 = p2 = 0

        while p1 < len(word1) and p2 < len(word2):
            word += word1[p1]
            word += word2[p2]
            p1 += 1
            p2 += 1

        if p1 < len(word1):
            word += word1[p1:]

        if p2 < len(word2):
            word += word2[p2:]

        return word

File: neetcode/1768-Merge-Strings-Alternately.ts
function mergeAlternately(word1: string, word2: string): string {
  const len1 = word1.length;
  const len2 = word2.length;
  const totalLength = len1 + len2;
  const res = new Array(totalLength);
  let index = 0, i = 0;

  while (i < len1 && i < len2) {
    res[index++] = word1[i];
    res[index++] = word2[i];
    i++;
  }

  while (i < len1) res[index++] = word1[i++];
  while (i < len2) res[index++] = word2[i++];

  return res.join('');
}